src/voice_registration.js

```diff
@@ -34,191 +35,615 @@ go.app = function() {
-                            .then(function(mama_id) {
-                                self.im.user.set_answer('mama_id', mama_id);
-                                return 'state_r03_receiver';
-                            });
+                        return 'state_pregnancy_status';
                     }
                 }
             });
         });
 
-        self.add('state_r03_receiver', function(name) {
+        // ChoiceState st-04
+        self.add('state_pregnancy_status', function(name) {
             var speech_option = '1';
             return new ChoiceState(name, {
-                question: $('Choose receiver'),
+                question: $('Pregnant or baby'),
                 helper_metadata: go.utils.make_voice_helper_data(
                     self.im, name, lang, speech_option),
                 choices: [
-                    new Choice('mother', $('Mother')),
-                    new Choice('other', $('Other'))
+                    new Choice('state_last_period_year', $('Pregnant')),
+                    new Choice('state_baby_birth_year', $('Baby'))
                 ],
-                next: 'state_r04_mom_state'
+                next: function(choice) {
+                    return choice.value;
+                }
             });
         });
 
-        self.add('state_r04_mom_state', function(name) {
+        // ChoiceState st-05
+        self.add('state_last_period_year', function(name) {
             var speech_option = '1';
             return new ChoiceState(name, {
-                question: $('Pregnant or baby'),
+                question: $('Last period?'),
                 helper_metadata: go.utils.make_voice_helper_data(
                     self.im, name, lang, speech_option),
                 choices: [
-                    new Choice('pregnant', $('Pregnant')),
-                    new Choice('baby', $('Baby'))
+                    new Choice('state_5A_period_month', $('This year')),
+                    new Choice('state_5B_period_month', $('Last year'))
                 ],
-                next: 'state_r05_birth_year'
+                next: function(choice) {
+                    return choice.value;
+                }
             });
         });
 
-        self.add('state_r05_birth_year', function(name) {
-            // TODO #6 Don't show next year for pregnancy in Jan / Feb
-            var speech_option;
-            var year_choices = [
-                new Choice('last_year', $('last_year')),
-                new Choice('this_year', $('this_year')),
-                new Choice('next_year', $('next_year'))
-            ];
-            if (self.im.user.answers.state_r04_mom_state === 'pregnant') {
-                choices = year_choices.slice(1,3);
-                speech_option = '1';
-            } else {
-                choices = year_choices.slice(0,2);
-                speech_option = '2';
-            }
+        // ChoiceState st-5A
+        self.add('state_5A_period_month', function(name) {
```

Comment:
Please replace the 5A here with `this_year` so it's clearer which state we're in